src/Features/Core/Portable/ExtractMethod/ExtractMethodMatrix.cs

```diff
@@ -82,9 +78,23 @@ static ExtractMethodMatrix()
                 }
             }
 
-            Contract.ThrowIfFalse(s_matrix.ContainsKey(key), $"Matrix does not contain Key '{key}'.");
+            if (s_matrix.TryGetValue(key, out variableStyle))
+            {
+                return true;
+            }
+
+            if (bestEffort)
+            {
+                // In best effort mode, even though we didn't know precisely what to do, we still
+                // allow the user to keep going, assuming that this variable is a very basic one.
+                variableStyle = VariableStyle.InputOnly;
+                return true;
+            }
 
-            return s_matrix[key];
+            // Some combination we didn't anticipate.  Can't do anything here.  Log the issue
+            // and bail out.
+            Logger.Log(FunctionId.Refactoring_ExtractMethod_UnknownMatrixItem, key.ToString());
```

Comment:
change this to NFW

Reply:
NFW's hang the process while the dump is being collected.  Is there a reason we need a memory dump, when it would likely suffice to just know about the id of something we missed?  Thanks!

Reply:
I would change this to NFW so that we have a way to get dump when we want to.

Reply:
Swithced to NFW.